src/ray/raylet/test/local_object_manager_test.cc

```diff
@@ -342,6 +388,23 @@ TEST_F(LocalObjectManagerTest, TestRestoreSpilledObject) {
   }
   worker_pool.io_worker_client->ReplyRestoreObjects(10);
   ASSERT_EQ(num_times_fired, 1);
+
+  // If the object wasn't spilled on the current node, it should request restoration to
+  // remote nodes.
+  ObjectID remote_object_id = ObjectID::FromRandom();
+  NodeID remote_node_id = NodeID::FromRandom();
+  manager.AsyncRestoreSpilledObject(remote_object_id, remote_node_id,
+                                    [&](const Status &status) {
+                                      ASSERT_TRUE(status.ok());
+                                      num_times_fired++;
+                                    });
+  // Make sure the callback wasn't called.
+  ASSERT_EQ(num_times_fired, 1);
```

Comment:
We actually do want the callback to fire though, right?

Reply:
You are right! This assert should've been moved to the end of the test as well (that's why it succeed). Fixed!